File: Lib/Telerik/Examples/QuickStart/PivotGrid/UnderlyingData/UnderlyingDataForm.vb
Imports System.Collections
Imports System.ComponentModel
Imports System.Text
Imports Telerik.WinControls
Imports Telerik.WinControls.UI

Namespace Telerik.Examples.WinControls.PivotGrid.UnderlyingData
	Partial Public Class UnderlyingDataForm
		Inherits Telerik.WinControls.UI.RadForm
		Public Sub New()
			InitializeComponent()

			Dim summaryItemQuantity As New GridViewSummaryItem()
			summaryItemQuantity.Name = "Quantity"
			summaryItemQuantity.Aggregate = GridAggregateFunction.Sum
			Dim summaryItemNet As New GridViewSummaryItem()
			summaryItemNet.Name = "Net"
			summaryItemNet.Aggregate = GridAggregateFunction.Sum

			Dim summaryRowItem As New GridViewSummaryRowItem()
			summaryRowItem.Add(summaryItemQuantity)
			summaryRowItem.Add(summaryItemNet)
			Me.radGridView1.SummaryRowsTop.Add(summaryRowItem)
		End Sub

		Friend Sub DisplayUnderlyingData(ByVal underlyingData As IEnumerable, ByVal owner As IWin32Window)
			Me.radGridView1.DataSource = underlyingData
			Me.ShowDialog(owner)
		End Sub
	End Class
End Namespace

File: Lib/Telerik/Examples/QuickStart/PivotGrid/UnderlyingData/UnderlyingDataForm.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using Telerik.WinControls;
using Telerik.WinControls.UI;

namespace Telerik.Examples.WinControls.PivotGrid.UnderlyingData
{
    public partial class UnderlyingDataForm : Telerik.WinControls.UI.RadForm
    {
        public UnderlyingDataForm()
        {
            InitializeComponent();

            GridViewSummaryItem summaryItemQuantity = new GridViewSummaryItem();
            summaryItemQuantity.Name = "Quantity";
            summaryItemQuantity.Aggregate = GridAggregateFunction.Sum;
            GridViewSummaryItem summaryItemNet = new GridViewSummaryItem();
            summaryItemNet.Name = "Net";
            summaryItemNet.Aggregate = GridAggregateFunction.Sum;

            GridViewSummaryRowItem summaryRowItem = new GridViewSummaryRowItem();
            summaryRowItem.Add(summaryItemQuantity);
            summaryRowItem.Add(summaryItemNet);
            this.radGridView1.SummaryRowsTop.Add(summaryRowItem);
        }

        internal void DisplayUnderlyingData(IEnumerable underlyingData, IWin32Window owner)
        {
            this.radGridView1.DataSource = underlyingData;
            this.ShowDialog(owner);
        }
    }
}
